src/extension.ts

```diff
@@ -38,27 +39,74 @@ export function activate(context: vscode.ExtensionContext) {
   // listen for configuration changes
   context.subscriptions.push(
     vscode.workspace.onDidChangeConfiguration((e) => {
-      const configChanged = e.affectsConfiguration('vimFindHighlight.charPrimaryColor') || e.affectsConfiguration('vimFindHighlight.charSecondaryColor') || e.affectsConfiguration('vimFindHighlight.charFontWeight') || e.affectsConfiguration('vimFindHighlight.enableUnderline');
+      const configChanged =
+        e.affectsConfiguration("vimFindHighlight.charPrimaryColor") ||
+        e.affectsConfiguration("vimFindHighlight.charSecondaryColor") ||
+        e.affectsConfiguration("vimFindHighlight.charFontWeight") ||
+        e.affectsConfiguration("vimFindHighlight.enableUnderline") ||
+        e.affectsConfiguration("vimFindHighlight.enableAutoHighlight");
 
       if (configChanged) {
         configureDecoration();
+        autoHighlight = configureAutoHighlight();
       }
     })
-  )
+  );
 
-  // when activating the extension. read the user settings file first
+  // when activating the extension. read the user settings file first and register the commands
   configureDecoration();
+  registerCommands(context);
 }
 
 const configureDecoration = () => {
   updateDecorationConfig();
   disposeCharDecoration();
 };
 
+const configureAutoHighlight = (): boolean => {
+  const settings = vscode.workspace.getConfiguration();
+  const autoHighlight = settings.get(
+    "vimFindHighlight.enableAutoHighlight"
+  ) as boolean;
+  vscode.window.showInformationMessage(autoHighlight.toString());
+  return autoHighlight;
+};
+
 const main = (cursorPos: number, currentLine: string) => {
   const toColor = charHighlighter.getCharHighlighting(currentLine, cursorPos);
   colorChars(toColor, decorationConfig);
 };
 
+const registerCommands = (context: vscode.ExtensionContext) => {
+  const toggleCommand = "vimFindHighlight.toggleAuthoHighlight";
+  context.subscriptions.push(
+    vscode.commands.registerCommand(toggleCommand, toggleAutoHighlight)
+  );
+
+  const highlightCommand = "vimFindHighlight.highlightCharacters";
+  context.subscriptions.push(
+    vscode.commands.registerCommand(highlightCommand, highlightCharacters)
+  );
+};
+
+const highlightCharacters = () => {
+  const line = getCurrentLine();
+  const cursorPos = getCursorPos();
+  if (line?.text.length && cursorPos) {
```

Comment:
I just tested this. If `cursorPos` is `0`. Nothing happens since Javascript treats it as false (what a language i know). So i would suggest to make the check for `undefined` explicit.
```suggestion
  const cursorPos = getCursorPos();
  if (line?.text.length && cursorPos !== undefined) {
```